crates/miden-lib/asm/kernels/transaction/lib/output_note.masm

```diff
@@ -41,6 +41,24 @@ export.get_assets_info
     # => [note_data_ptr, num_assets]
 
     # get the assets commitment
-    exec.note::compute_output_note_assets_commitment
+    dup exec.note::compute_output_note_assets_commitment
+    # => [ASSETS_COMMITMENT, note_data_ptr, num_assets]
+
+    # get the start and the end pointers of the asset data
+    movup.4 exec.memory::get_output_note_asset_data_ptr
+    # => [assets_data_ptr, ASSETS_COMMITMENT, num_assets]
+
+    dup dup.6 mul.4 add
+    # => [assets_end_ptr, assets_start_ptr, ASSETS_COMMITMENT, num_assets]
```

Comment:
When we hash assets, we round them up to a double-word boundary, right? If so, should we not do the same for `asset_end_ptr`? Maybe this would solve the issue you were describing in the other comment?